README.md

```diff
@@ -1,8 +1,362 @@
+As a result, we recommend anyone using CCCL from GitHub to use CMake to integrate CCCL into your project. 
+For a complete example of how to do this using CMake Package Manager see [our example project](examples/example_project). 
+
+Other build systems should work, but we only test CMake.
+We welcome contributions that would simplify the process of integrating CCCL into other build systems.
+
+### Contributors
+
+Contributor guide coming soon!
+
+## Platform Support
+
+**Objective:** This section describes where users can expect CCCL to compile and run successfully. 
+
+In general, CCCL should work everywhere the CUDA Toolkit is supported, however, the devil is in the details. 
+The sections below describe the details of our support for different versions of the CUDA Toolkit, host compilers, and C++ dialects.
+Furthermore, we describe our testing strategy to be confident that CCCL works everywhere it should. 
+
+### CUDA Toolkit (CTK) Compatibility 
+
+**Summary:**
+- The latest version of CCCL is backwards compatible with the current and preceding CTK major version series
+- CCCL is never forwards compatible with any version of the CTK. Always use the same or newer than what is included with your CTK. 
+- Minor version CCCL upgrades won't break existing code, but new features may not support all CTK versions
+
+The CCCL team encourages users to ["live at head"](https://www.youtube.com/watch?v=tISy7EJQPzI) and always use the newest version of CCCL from GitHub to take advantage of the latest improvements.
+
+We want to enable users to upgrade the version of CCCL they are using without upgrading their entire CUDA Toolkit. 
+Therefore, CCCL is backwards compatible with the latest patch release of every minor CTK release within two major version series: current and preceding. 
+When a new major CTK is released, we drop support for the oldest version.
+
+| CCCL Version  | Supports CUDA Toolkit Version  |
+|---------------|-------------------------------|
+| 2.x           | 11.1 - 11.8, 12.x (only latest patch releases) |
+| 3.x (Future)  | 12.x, 13.x  (only latest patch releases)       |
+
+[Well-behaved code](#compatibility-guidelines) using the latest CCCL should compile and run successfully with any supported CTK version. 
+Exceptions may occur for new features that depend on new CTK features, so those features would not work on older versions of the CTK. 
+For example, C++20 support was not added to `nvcc` until CUDA 12.0, so CCCL features that depend on C++20 would not work with CTK 11.x. 
+
+We want to enable users to bring a newer version of CCCL to an older CTK, but not vice versa.
+This means you cannot bring an older version of CCCL to a newer CTK.
+In other words, **CCCL is never forwards compatible with the CUDA Toolkit.** 
+
+The table below summarizes compatibility of the CTK and CCCL:
```

Comment:
Q: Do we have a table somewhere showing the version map from existing CTK 11.x, 12.x to CCCL versions? If so, perhaps we can link it here as a reference?